### PR TITLE
Add list of previous work that contributed to the science.

### PR DESCRIPTION
**Pull request by msporny**
This PR attempts to address issue #19 by adding a history of previous work that contributed to the science that this specification is based upon.

***
<a href="https://pr-preview.s3.amazonaws.com/msporny/rdf-canon/pull/199.html" title="Last updated on Mar 1, 2024, 5:26 PM UTC (56e21f9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/rdf-canon/199/9cf1297...msporny:56e21f9.html" title="Last updated on Mar 1, 2024, 5:26 PM UTC (56e21f9)">Diff</a>

### CODE REVIEW

File: spec/index.html [hunk 1]
```diff
@@ -3775,6 +3775,107 @@ <h2>Acknowledgements</h2>
...
+      (eds) The Semantic Web: Trends and Challenges. ESWC 2014. Lecture Notes in
+      Computer Science, vol 8465. Springer, Cham.
+      https://doi.org/10.1007/978-3-319-07443-6_11 [[eswc2014Kasten]]
+    </li>
+    <li>
+      Aidan Hogan. 2017. Canonical Forms for Isomorphic and Equivalent RDF
+      Graphs: Algorithms for Leaning and Labelling Blank Nodes. ACM Trans. Web
+      0, 0, Article 00 (January 2017), 61 pages. [[Hogan-Canonical-RDF]]
+    </li>
+    <li>
```
Comment (dlongley) [inline spec/index.html:3873]: Not sure if the formatting is correct, reference taken [from here](https://www.w3.org/TR/rch-explainer/#informative-references) and rewritten to try and match the style used here. It was respec-ified in the other place.

```suggestion
    </li>
    <li>
      Rachel Arnold, Dave Longley. 2020. RDF Dataset Normalization. Report
      submitted to the W3C Credentials Community Group mailing list.
      https://lists.w3.org/Archives/Public/public-credentials/2021Mar/att-0220/RDFDatasetCanonicalization-2020-10-09.pdf [[arnold-longley-2020]]
    </li>
    <li>
```

Comment (gkellogg) [reply spec/index.html:3873]: This is effectively the same as the `[CCG-RDC-FINAL]` reference. I'm fine for spelling it out, but be may want to make the two different references consistent.

The source for CCG-RDC-FINAL is in common/biblio.js.

**Review by gkellogg — APPROVED** [review]

**Review by gkellogg — APPROVED** [review]
My earlier comment was hidden:

This (The CG report reference) is effectively the same as the [CCG-RDC-FINAL] reference. I'm fine for spelling it out, but be may want to make the two different references consistent.

The source for CCG-RDC-FINAL is in common/biblio.js.

**Comment by davidlehn** [comment]
- It does bother me a little bit that there is a nice structured localBiblio/specref system here and respec can render that uniformly and nicely, and yet these are free form text.  is there is a way to call the render function for a specific entries so this could be a presentation that aligns with the references style but in a simple list?
- The above might be hard.  I'd suggest at least normalizing style for things like dates as "2010" or "(2010)", and a trailing period or not.

**Comment by peacekeeper** [comment]
Looks good to me..

**Comment by iherman** [comment]
ON the remark of @davidlehn (https://github.com/w3c/rdf-canon/pull/199#issuecomment-1970293890) while it is correct that a proper, scholarly reference system would be nice, let us not spend our time finding one and merging with respec...

Alternatively, we could feed all those references into a system like Zotero, and then spit out a properly formatted references. But that would take probably too much time for a one off occasion

I agree that the references should be uniform, though. IEEE has a document on citation format: https://ieeeauthorcenter.ieee.org/wp-content/uploads/IEEE-Reference-Guide.pdf, and I am sure that Springer, Elsevier or others have the same. I would think just pick one of those and reformat all references (yes, by hand...) is the shortest and the quickest way of solving this.

**Comment by iherman** [comment]
- I realize that the reference list is a bit arbitrary; I am not sure where it comes from. For this community, I wonder whether it is important to add the old, graph-theory references that were written even pre-RDF days/ I would think that we can safely prune the list by removing the first part up until the entry from TimBL and DanC, as well as removing the entry from Gross and Yellen. Just concentrate on specifically RDF & Co papers is enough.
- The references to Adrian sound a bit strange ("Web 0, 0, Article 00"). The [explainer document's references](https://www.w3.org/2022/07/rch-wg-charter/explainer.html#references) might help.
- I was wondering about _adding_ the [paper of Jeremy, Pat, Chris, and Patrick](https://www.sciencedirect.com/science/article/abs/pii/S1570826805000235?via%3Dihub). One thing we did here compared to many other references is to create an algorithm for _datasets_ and not only _graphs_, and that paper is one of the first that introduced the notion of named graph with a specific section devoted to signatures.

**Comment by yamdan** [comment]
According to @iherman 's https://github.com/w3c/rdf-canon/pull/199#issuecomment-1970596570 , I tried formatting the list made by Manu in my Zotero into the Springer-LNCS format. Some of the references are failing to fetch metadata because they are not scholarly articles. If this LNCS format is okay, I can try making the other ones manually. What do you think?

- Corneil, D., Goldberg, M.: A non-factorial algorithm for canonical numbering of a graph. Journal of Algorithms. 5, 345–362 (1984). https://doi.org/10.1016/0196-6774(84)90015-4.
- Weisfeiler, B.: On Construction and Identification of Graphs. Springer, Berlin, Heidelberg (1976). https://doi.org/10.1007/BFb0089374.
- Babai, L., Erdős, P., Selkow, S.M.: Random Graph Isomorphism. SIAM J. Comput. 9, 628–635 (1980). https://doi.org/10.1137/0209047.
- Hoffmann, C.M. ed: Group-Theoretic Algorithms and Graph Isomorphism. Springer, Berlin, Heidelberg (1982). https://doi.org/10.1007/3-540-11493-9.
- Corneil, D.G., Gotlieb, C.C.: An Efficient Algorithm for Graph Isomorphism. J. ACM. 17, 51–64 (1970). https://doi.org/10.1145/321556.321562.
- Sutcliffe, G., Suttner, C.: The TPTP Problem Library. Journal of Automated Reasoning. 21, 177–203 (1998). https://doi.org/10.1023/A:1005806324129.
- Grohe, M.: Isomorphism testing for embeddable graphs through definability. In: Proceedings of the thirty-second annual ACM symposium on Theory of computing. pp. 63–72. ACM, Portland Oregon USA (2000). https://doi.org/10.1145/335305.335313.
- (W.W. Tsang)
- (Tim Berners-Lee and Dan Connolly)
- (Jeremy J. Carroll)
- Carroll, J.J.: Signing RDF Graphs. In: Fensel, D., Sycara, K., and Mylopoulos, J. (eds.) The Semantic Web - ISWC 2003. pp. 369–384. Springer, Berlin, Heidelberg (2003). https://doi.org/10.1007/978-3-540-39718-2_24.
- Gross, J.L., Yellen, J., Zhang, P.: Handbook of Graph Theory, Second Edition. Chapman & Hall/CRC (2013).
- (C. Sayers, A. Karp)
- Tummarello, G., Morbidoni, C., Bachmann-Gmür, R., Erling, O.: RDFSync: Efficient Remote Synchronization of RDF Models. In: Aberer, K., Choi, K.-S., Noy, N., Allemang, D., Lee, K.-I., Nixon, L., Golbeck, J., Mika, P., Maynard, D., Mizoguchi, R., Schreiber, G., and Cudré-Mauroux, P. (eds.) The Semantic Web. pp. 537–551. Springer, Berlin, Heidelberg (2007). https://doi.org/10.1007/978-3-540-76298-0_39.
- Oren, E., Delbru, R., Catasta, M., Cyganiak, R., Stenzhorn, H., Tummarello, G.: Sindice.com: a document-oriented lookup index for open linked data. International Journal of Metadata, Semantics and Ontologies. 3, 37–52 (2008). https://doi.org/10.1504/IJMSO.2008.021204.
- Fisteus, J.A., Fernández García, N., Sánchez Fernández, L., Delgado Kloos, C.: Hashing and canonicalizing Notation 3 graphs. Journal of Computer and System Sciences. 76, 663–685 (2010). https://doi.org/10.1016/j.jcss.2010.01.003.
- Kasten, A., Scherp, A., Schauß, P.: A Framework for Iterative Signing of Graph Data on the Web. In: Presutti, V., d’Amato, C., Gandon, F., d’Aquin, M., Staab, S., and Tordai, A. (eds.) The Semantic Web: Trends and Challenges. pp. 146–160. Springer International Publishing, Cham (2014). https://doi.org/10.1007/978-3-319-07443-6_11.
- Hogan, A.: Canonical Forms for Isomorphic and Equivalent RDF Graphs: Algorithms for Leaning and Labelling Blank Nodes. ACM Trans. Web. 11, 22:1-22:62 (2017). https://doi.org/10.1145/3068333.
- (Rachel Arnold, Dave Longley)
- Tomaszuk, D., Glab, S., Turobos, F., Pawlik, T., Kuzinski, D., Sopek, M.: Interwoven Hash of Vicious Circle Free Graph. In: 2022 IEEE International Conference on Blockchain (Blockchain). pp. 449–454. IEEE, Espoo, Finland (2022). https://doi.org/10.1109/Blockchain55522.2022.00069.

**Comment by iherman** [comment]
@yamdan, thank you. That looks perfect. Springer's LNCS format is as good as any, the only point that we should be consistent.

**Comment by msporny** [comment]
> This (The CG report reference) is effectively the same as the [CCG-RDC-FINAL] reference. I'm fine for spelling it out, but be may want to make the two different references consistent.

One of them is the proof of correctness and the other is the CCG-RDC-FINAL document -- two documents with very different content, unless I'm missing something? 

> The above might be hard. I'd suggest at least normalizing style for things like dates as "2010" or "(2010)", and a trailing period or not.

Yes, I contemplated added Specref entries for every one of those papers... it would take hours to do that (which I don't have). Others are welcome to add all of the entries to specref (but again, doing that for just this spec doesn't seem like a good use of time).

So, it's not easy, no... the references were collected over the years from citations in papers that were published in journals (each journal used a slightly different format). I just used the format used in each journal.

> I tried formatting the list made by Manu in my Zotero into the Springer-LNCS format.

Thank you for doing that @yamdan ... any chance you could raise those changes as a change set against this PR? I won't have the spare cycles in the coming weeks to go back through and re-format everything.

I'm happy for someone else to take over the reformatting for this PR at this point... it's all editorial work.

**Review by dlongley — APPROVED** [review]

**Review by philarcher — CHANGES_REQUESTED** [review]
I'm perfectly happy with the list of previous work - and I am glad it's in the spec. It shows that this hasn't come out of nowhere and has a long pedigree. One minor gripe - and my apologies if this seems more than a little self-serving - but there is a small group of people who turned up at more or less every meeting who discussed the spec as it evolved and got it through the Rec Track. I'd say they are: Pierre-Antoine, Ivan, Gregg, David, Dave, Kazue, Manu, Ted, Dan, Markus, self. Of course many of those people have, rightly, been credited already but, maybe there's a little space somewhere?

**Comment by gkellogg** [comment]
> One minor gripe - and my apologies if this seems more than a little self-serving - but there is a small group of people who turned up at more or less every meeting who discussed the spec as it evolved and got it through the Rec Track. I'd say they are: Pierre-Antoine, Ivan, Gregg, David, Dave, Kazue, Manu, Ted, Dan, Markus, self. Of course many of those people have, rightly, been credited already but, maybe there's a little space somewhere?

This would usually be added to the Acknowledgements section, but as this is the editors acknowledging others who were helpful in the work. We could add to the end of (or after) the first paragraph the specific contributions of the chairs, Pierre-Antoine, Ivan, Dave, Kazue, Manu, and Ted. Note that Dave and Manu are already called out as participants of the CG.

**Comment by gkellogg** [comment]
> Thank you for doing that @yamdan ... any chance you could raise those changes as a change set against this PR? I won't have the spare cycles in the coming weeks to go back through and re-format everything.

I can attempt to make those changes by pushing into your branch (which I believe I can do). It's easier if the branch is on this repo, but possible I believe.

> I'm happy for someone else to take over the reformatting for this PR at this point... it's all editorial work.

Thanks for getting this far. We still may want to trim the references, as @iherman suggested.

File: spec/index.html [hunk 2]
```diff
@@ -3775,6 +3788,149 @@ <h2>Acknowledgements</h2>
...
+      <a href="https://doi.org/10.1007/BFb0089374">https://doi.org/10.1007/BFb0089374</a>.
+    </li>
+    <li>
+      Babai, L., Erdős, P., Selkow, S.M.:
+      Random Graph Isomorphism.
+      SIAM J. Comput. 9, 628–635 (1980).
+      <a href="https://doi.org/10.1137/0209047">https://doi.org/10.1137/0209047</a>.
+    </li>
+    <li>
+      Hoffmann, C.M. ed: Group-Theoretic Algorithms and Graph Isomorphism.
```
Comment (davidlehn) [inline spec/index.html:3823]:
```suggestion
      Hoffmann, C.M. ed:
      Group-Theoretic Algorithms and Graph Isomorphism.
```

File: spec/index.html [hunk 3]
```diff
@@ -3775,6 +3788,149 @@ <h2>Acknowledgements</h2>
...
+    <li>
+      Carroll J.J., Bizer C., Hayes P., Stickler P.
+      Named graphs
+      J Web Semant, 3 (4) (2005), pp. 247-267, 10.1016/j.websem.2005.09.001.
+      World Wide Web Conference 2005——Semantic Web Track
+      <a href="https://www.sciencedirect.com/science/article/pii/S1570826805000235">https://www.sciencedirect.com/science/article/pii/S1570826805000235</a>.
+    </li>
+    <li>
+      Tummarello, G., Morbidoni, C., Bachmann-Gmür, R., Erling, O.: RDFSync:
+      Efficient Remote Synchronization of RDF Models.
```
Comment (davidlehn) [inline spec/index.html:3888]:
```suggestion
      Tummarello, G., Morbidoni, C., Bachmann-Gmür, R., Erling, O.:
      RDFSync: Efficient Remote Synchronization of RDF Models.
```

**Review by davidlehn — COMMENTED** [review]
Local biblio and respec rendering may have been easier.

File: spec/index.html [hunk 4]
```diff
@@ -3775,6 +3788,149 @@ <h2>Acknowledgements</h2>
...
+      In: Fensel, D., Sycara, K., and Mylopoulos, J. (eds.)
+      The Semantic Web - ISWC 2003. pp. 369–384.
+      Springer, Berlin, Heidelberg (2003).
+      <a href="https://doi.org/10.1007/978-3-540-39718-2_24">https://doi.org/10.1007/978-3-540-39718-2_24</a> [[HPL-2003-142]].
+    </li>
+    <li>
+      Gross, J.L., Yellen, J., Zhang, P.:
+      Handbook of Graph Theory, Second Edition.
+      Chapman & Hall/CRC (2013).
+    </li>
```
[the remark this answers is not in the document]
Comment (gkellogg) [reply spec/index.html:3871]: I'll go through the ordering after merging the suggestions.

File: spec/index.html [hunk 5]
```diff
@@ -3775,6 +3788,149 @@ <h2>Acknowledgements</h2>
...
+      Tech. Rep. HPL-2003-235 (R. 1),
+      Hewlett Packard Laboratories, 2004.
+      <a href="https://www.hpl.hp.com/techreports/2003/HPL-2003-235R1.html">https://www.hpl.hp.com/techreports/2003/HPL-2003-235R1.html</a>.
+    </li>
+    <li>
+      Carroll J.J., Bizer C., Hayes P., Stickler P.
+      Named graphs
+      J Web Semant, 3 (4) (2005), pp. 247-267, 10.1016/j.websem.2005.09.001.
+      World Wide Web Conference 2005——Semantic Web Track
+      <a href="https://www.sciencedirect.com/science/article/pii/S1570826805000235">https://www.sciencedirect.com/science/article/pii/S1570826805000235</a>.
```
[the remark this answers is not in the document]
Comment (gkellogg) [reply spec/index.html:3884]: That was taken from the suggested citation reference, but no problem with using your suggestion.

File: spec/index.html [hunk 6]
```diff
@@ -3775,6 +3788,150 @@ <h2>Acknowledgements</h2>
...
+      Computing the digest of an RDF graph,
+      Tech. Rep. HPL-2003-235 (R. 1),
+      Hewlett Packard Laboratories (2004).
+      <a href="https://www.hpl.hp.com/techreports/2003/HPL-2003-235R1.html">https://www.hpl.hp.com/techreports/2003/HPL-2003-235R1.html</a>.
+    </li>
+    <li>
+      Carroll, J.J., Bizer, C., Hayes, P., Stickler, P.:
+      Named graphs.
+      Journal of Web Semantics, 3 (4), pp. 247-267,
+      World Wide Web Conference 2005——Semantic Web Track  (2005).
```
Comment (davidlehn) [inline spec/index.html:3879]: fix space.  is two dashes correct?
```suggestion
      World Wide Web Conference 2005——Semantic Web Track (2005).
```

Comment (gkellogg) [reply spec/index.html:3879]: It was copied from the “how to cite this paper” section in the reference. Of course, we can do as we see fit.

Comment (davidlehn) [reply spec/index.html:3879]: Double whatever dashes those are looks silly long in my browser, I changed it above to one.  Others feel free to comment on appropriate dash usage.

File: spec/index.html [hunk 7]
```diff
@@ -3775,6 +3788,150 @@ <h2>Acknowledgements</h2>
...
+      In: Presutti, V., d’Amato, C., Gandon, F., d’Aquin, M., Staab, S., and Tordai, A. (eds.)
+      The Semantic Web: Trends and Challenges. pp. 146–160.
+      Springer International Publishing, Cham (2014).
+      <a href="https://doi.org/10.1007/978-3-319-07443-6_11">https://doi.org/10.1007/978-3-319-07443-6_11</a> [[eswc2014Kasten]].
+    </li>
+    <li>
+      Hogan, A.:
+      [[[Hogan-Canonical-RDF]]].
+      ACM Trans. Web. 11, 22:1-22:62 (2017).
+      <a href="https://doi.org/10.1145/3068333">https://doi.org/10.1145/3068333</a> [[Hogan-Canonical-RDF]].
```
Comment (davidlehn) [inline spec/index.html:3919]: Also paywalled, I think.  But the specref link goes to authors pdf link.  Good to leave DOI link here?

File: spec/index.html [hunk 8]
```diff
@@ -3775,6 +3788,150 @@ <h2>Acknowledgements</h2>
...
+      RDF Dataset Normalization.
+      Report submitted to the W3C Credentials Community Group mailing list (2020).
+      <a href="https://lists.w3.org/Archives/Public/public-credentials/2021Mar/att-0220/RDFDatasetCanonicalization-2020-10-09.pdf">https://lists.w3.org/Archives/Public/public-credentials/2021Mar/att-0220/RDFDatasetCanonicalization-2020-10-09.pdf</a>.
+    </li>
+    <li>
+      Tomaszuk, D., Glab, S., Turobos, F., Pawlik, T., Kuzinski, D., Sopek, M.:
+      Interwoven Hash of Vicious Circle Free Graph.
+      In: 2022 IEEE International Conference on Blockchain (Blockchain). pp. 449–454. IEEE,
+      Espoo, Finland (2022).
+      <a href="https://doi.org/10.1109/Blockchain55522.2022.00069">https://doi.org/10.1109/Blockchain55522.2022.00069</a>.
```
Comment (davidlehn) [inline spec/index.html:3932]: Also paywalled.  Hard to find elsewhere.  Resolve comment if this is fine.

File: spec/index.html [hunk 9]
```diff
@@ -3775,6 +3788,150 @@ <h2>Acknowledgements</h2>
...
+      <a href="https://doi.org/10.1145/3068333">https://doi.org/10.1145/3068333</a> [[Hogan-Canonical-RDF]].
+    </li>
+    <li>
+      Arnold, R., Longley, D.:
+      RDF Dataset Normalization.
+      Report submitted to the W3C Credentials Community Group mailing list (2020).
+      <a href="https://lists.w3.org/Archives/Public/public-credentials/2021Mar/att-0220/RDFDatasetCanonicalization-2020-10-09.pdf">https://lists.w3.org/Archives/Public/public-credentials/2021Mar/att-0220/RDFDatasetCanonicalization-2020-10-09.pdf</a>.
+    </li>
+    <li>
+      Tomaszuk, D., Glab, S., Turobos, F., Pawlik, T., Kuzinski, D., Sopek, M.:
```
Comment (yamdan) [inline spec/index.html:3928]: It seems that a few diacritics were removed when I processed it through Zotero. I have restored them.
```suggestion
      Tomaszuk, D., Głąb, S., Turoboś, F., Pawlik, T., Kuziński, D., Sopek, M.:
```

**Review by yamdan — COMMENTED** [review]
Thank you @gkellogg for updating the list. I found that a few diacritics were removed when I processed it through Zotero...

File: spec/index.html (hunk 6 above)
Comment (yamdan) [inline spec/index.html:3879]: To reflect the above suggestion:
```suggestion
      Journal of Web Semantics. 3, 247–267 (2005).
```

**Review by iherman — COMMENTED** [review]
It looks fine to me now, except for my remark on pruning. I won't lie down the road on this, but it looks unnecessary, maybe even presumptuous to me here (there are tons of papers on graph theory, why picking just these few at this point?).

File: spec/index.html (hunk 7 above)
Comment (gkellogg) [reply spec/index.html:3918]: The DOI is more "official", so it's good to use as a primary reference. In this case, we had a bib entry, so adding it was easy.

**Review by iherman — APPROVED** [review]

File: spec/index.html [hunk 10]
```diff
@@ -3746,7 +3746,7 @@ <h2>Acknowledgements</h2>
     
   <p>The editors would also like to thank
     the chairs of the Working Group Phil Archer and Markus Sabadello
```
Comment (TallTed) [inline spec/index.html:3748]:
```suggestion
    the chairs of the Working Group, Phil Archer and Markus Sabadello,
```

File: spec/index.html [hunk 11]
```diff
@@ -3746,7 +3746,7 @@ <h2>Acknowledgements</h2>
...
-    and specific members the Working Group who's active contributions
+    and specific members the Working Group whose active contributions
```
Comment (TallTed) [inline spec/index.html:3749]:
```suggestion
    and specific members of the Working Group whose active contributions
```

File: spec/index.html (hunk 10 above)
Comment (TallTed) [reply spec/index.html:3748]: or
```suggestion
    the chairs of the Working Group — Phil Archer and Markus Sabadello —
```

File: spec/index.html [hunk 12]
```diff
@@ -3743,6 +3743,19 @@ <h2>Acknowledgements</h2>
...
+  <p>The editors would also like to thank
+    the chairs of the Working Group Phil Archer and Markus Sabadello
+    and specific members the Working Group whose active contributions
+    were critical in completing this work:
+    Pierre-Antoine Champin,
+    Ivan Herman,
+    David Lehn,
+    Kazue Sako,
+    Manu Sporny, and
+    Ted Thibodeau.
```
Comment (TallTed) [inline spec/index.html:3756]:
```suggestion
    Ted Thibodeau Jr.
```

File: spec/index.html [hunk 13]
```diff
@@ -3775,6 +3788,149 @@ <h2>Acknowledgements</h2>
     not necessarily reflect the position or the policy of the U.S. Government and no
     official endorsement should be inferred.
   </p>
+
+  <p>
+    The Working Group acknowledges that the success of this specification is
+    dependent on a long history of work performed over multiple decades in both
+    academia and industry. We thank the individuals that iterated on the science
```
Comment (TallTed) [inline spec/index.html:3795]:
```suggestion
    academia and industry. We thank the individuals who iterated on the science
```

**Review by TallTed — CHANGES_REQUESTED** [review]
A few small things

**Review by TallTed — APPROVED** [review]

**Pull request merged**